File: Napakalaki/src/napakalaki/Treasure.java

package napakalaki;

/**
 *
 * @author alex
 */
public class Treasure {
    
    // DECLARACIÓN DE VARIABLES
    
    private String name;
    private int bonus;
    private TreasureKind type;
    
    // CONSTRUCTOR

    public Treasure(String name, int bonus, TreasureKind type) {
        this.name = name;
        this.bonus = bonus;
        this.type = type;
    }
    
    // GETTERS

    public String getName() {
        return name;
    }

    public int getBonus() {
        return bonus;
    }

    public TreasureKind getType() {
        return type;
    }

    @Override
    public String toString() {
        return "Tesoro: " + name + ", bonus=" + bonus + ", tipo=" + type;
    }
    
}

File: NapakalakiRuby/pkg/NapakalakiRuby-0.0.1/lib/treasure.rb
# coding: utf-8

require_relative 'treasure_kind.rb'

module Napakalaki
  
  class Treasure
    
    # CONSULTORES
    
    def getType
      @type
    end
    
    def getBonus
      @bonus
    end
    
    def getName
      @name
    end
    
    # INICIALIZADOR
    
    def initialize(n, bonus, t)
      @name=n
      @bonus=bonus
      @type=t
    end
    
    # Método to_string

    def to_s
      "Tesoro: #{@name} \tBonus: #{@bonus} \tTipo: #{@type}"
    end
  
  end

end